### PR TITLE
Use varint encoding for range ids in keys.

### PR DESCRIPTION
**Pull request by petermattis**
The varint encoding and decoding is significantly faster than the
numeric encoding.

### CODE REVIEW

File: storage/engine/encoding.cc [hunk 1]
```diff
@@ -24,16 +24,37 @@ const unsigned char kEscape      = 0x00;
 const unsigned char kEscapedTerm = 0x01;
 const unsigned char kEscapedNul  = 0xff;
 
+template <typename T>
+bool DecodeVarint(rocksdb::Slice* buf, T* value) {
+  if (buf->empty()) {
+    return false;
+  }
+  int len = (*buf)[0];
+  if ((len + 1) > buf->size()) {
```
Comment (bdarnell) [inline storage/engine/encoding.cc]: Should we also guard against `len > sizeof(T)` here?

**Comment by bdarnell** [comment]
LGTM

File: storage/engine/db.cc [hunk 2]
```diff
@@ -59,11 +59,13 @@ struct DBSnapshot {
 
 namespace {
 
-// NOTE: these constants must be kept in sync with the values
-// in storage/engine/keys.go.
-const rocksdb::Slice kKeyLocalRangeIDPrefix("\x00\x00\x00i", 4);
-const rocksdb::Slice kKeyLocalRangeKeyPrefix("\x00\x00\x00k", 4);
-const rocksdb::Slice kKeyLocalResponseCacheSuffix("\x00res-", 5);
+// NOTE: these constants must be kept in sync with the values in
+// storage/engine/keys.go. Both kKeyLocalRangeIDPrefix and
```
Comment (tschottdorf) [inline storage/engine/db.cc]: Is there a corresponding comment in `storage/engine/keys.go`?

Comment (petermattis) [inline storage/engine/db.cc]: No. I'll add one.

File: storage/engine/encoding.cc (hunk 1 above)
Comment (petermattis) [inline storage/engine/encoding.cc]: Doesn't hurt. Note that this code was pulled verbatim from the viewfinder sources.

**Pull request merged**